### PR TITLE
Add Moco service / Add CMD capability / Change ENTRYPOINT to EXECFORM from SHELLFORM

### PR DESCRIPTION
**Pull request by anivon**
- Add Moco Service (with simple example)
- Add CMD capability
   - [Change](https://github.com/Ph3nol/Docker-Arch/pull/7/files#diff-c4801a92f7c0ad99c8ae3ae6e9b40a92R33)
   - [Doc](https://docs.docker.com/engine/reference/builder/#cmd)
- Change ENTRYPOINT to EXECFORM from SHELLFORM
   - [Change](https://github.com/Ph3nol/Docker-Arch/pull/7/files#diff-687b03c0426c2b1a20ba8b9f54b86e3cR67)
   - [Doc](https://docs.docker.com/engine/reference/builder/#entrypoint)

### CODE REVIEW

File: src/Application/Service/MocoService.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getOptionsResolver(): Options
+    {
+        $resolver = new OptionsResolver();
+
+        $resolver->setRequired('mock_filename');
+        $resolver->setAllowedTypes('mock_filename', 'string');
```
Comment (anivon) [inline src/Application/Service/MocoService.php:22]: In the future, why not handle multiple mock files. Or keep an single mock container approach.
TBD

Comment (Ph3nol) [reply src/Application/Service/MocoService.php:22]: Good idea! 👍

**Comment by Ph3nol** [comment]
Can you add link capability for `moco` into PhpServcice?
https://github.com/Ph3nol/Docker-Arch/blob/master/src/Application/Service/PhpService.php#L61

File: examples/simple-moco-project/mocks/moco.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[
+    {
+        "response": {
+            "text": "Hi, I'm the moco container :D"
```
Comment (Ph3nol) [inline examples/simple-moco-project/mocks/moco.json]: What do you think about `Hi, I'm a unicorn, BRO.`? 😄

Comment (anivon) [reply examples/simple-moco-project/mocks/moco.json]: I love this idea Bro, I'll do it soon !!! :)

File: src/Application/Service/MocoService.php (hunk 1 above)
Comment (Ph3nol) [reply src/Application/Service/MocoService.php:22]: What about using this:

``` json
[
    {
        "include" : "file1.json"
    },
    {
        "include" : "file2.json"
    }
]
```

Seems to be implemented for this.

https://github.com/dreamhead/moco/blob/master/moco-doc/global-settings.md

File: src/Application/Resources/views/Base/Service/Dockerfile.twig [hunk 3]
```diff
@@ -64,7 +64,13 @@ RUN {{ command|raw }}
 {% if dockerContainer.entryPoint %}
 
 # Entry point.
-ENTRYPOINT {{ dockerContainer.entryPoint }}
+ENTRYPOINT ["{{ dockerContainer.entryPoint }}"]
```
Comment (Ph3nol) [inline src/Application/Resources/views/Base/Service/Dockerfile.twig:67]: 👍

File: src/Application/DockerContainer/MocoDockerContainer.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        parent::init();
+
+        $this->setFrom('rezzza/docker-moco');
+        $this->applyWebServiceConfiguration();
+        $this->setMaintainer('Alexis NIVON <anivon@alexisnivon.fr>');
+
+        $mocoServerPort = $this->addEnvPort('MOCO_SERVER', ['from' => '8888', 'to' => self::MOCO_INTERNAL_PORT]);
+        $this->getService()->addUIAccess([
+            'port' => $mocoServerPort['from'],
+            'label' => 'Moco base URL',
```
Comment (Ph3nol) [inline src/Application/DockerContainer/MocoDockerContainer.php]: This UI entry is added to Moco Service, what about naming it `Base URL` instead?

File: src/Application/Service/MocoService.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+class MocoService extends AbstractService
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getOptionsResolver(): Options
+    {
+        $resolver = new OptionsResolver();
+
+        $resolver->setRequired('mock_filename');
```
Comment (Ph3nol) [inline src/Application/Service/MocoService.php:21]: What about `mock_file_path` name instead?

File: src/Application/Service/MocoService.php (hunk 1 above)
Comment (anivon) [reply src/Application/Service/MocoService.php:22]: In fact, I doesn't think to multiple files for one "service" (one port) but I think about multiple files, each representing an independent service on a specific port.

File: src/Application/DockerContainer/MocoDockerContainer.php (hunk 4 above)
Comment (anivon) [reply src/Application/DockerContainer/MocoDockerContainer.php]: :+1:

File: src/Application/Service/MocoService.php (hunk 5 above)
Comment (anivon) [reply src/Application/Service/MocoService.php:21]: My POV : `file_path` suggests we can pass absolute or relative path. 
But currently, data are always put into `/apps/..`.

File: src/Application/Service/MocoService.php (hunk 1 above)
Comment (Ph3nol) [reply src/Application/Service/MocoService.php:22]: OK! My bad, I didn't understand your point of view!

**Comment by Ph3nol** [comment]
Thanks for your contrib BRO.

**Pull request merged**